src/topology/bornology/order.lean

```diff
@@ -0,0 +1,190 @@
+⟨λ b₁ b₂, ⟨@cobounded α b₁ ⊔ @cobounded α b₂, sup_le b₁.le_cofinite b₂.le_cofinite⟩⟩
+
+instance : has_sup (bornology α) :=
+⟨λ b₁ b₂, ⟨@cobounded α b₁ ⊓ @cobounded α b₂, inf_le_of_left_le b₁.le_cofinite⟩⟩
+
+instance : has_bot (bornology α) :=
+⟨bornology.cofinite⟩
+
+instance : has_top (bornology α) :=
+⟨⟨⊥, bot_le⟩⟩
+
+instance : lattice (bornology α) :=
+{ le_sup_left := λ b₁ b₂, by {rw [bornology.le_def], exact inf_le_left},
+  le_sup_right := λ b₁ b₂, by {rw [bornology.le_def], exact inf_le_right},
+  sup_le := λ b₁ b₂ b₃ h₁₃ h₂₃, by {rw [bornology.le_def] at *, exact le_inf h₁₃ h₂₃},
+  inf_le_left := λ b₁ b₂, by {rw [bornology.le_def], exact le_sup_left},
+  inf_le_right := λ b₁ b₂, by {rw [bornology.le_def], exact le_sup_right},
+  le_inf := λ b₁ b₂ b₃ h₁₃ h₂₃, by {rw [bornology.le_def] at *, exact sup_le h₁₃ h₂₃},
+  ..bornology.partial_order,
+  ..bornology.has_inf,
+  ..bornology.has_sup }
+
+instance : complete_lattice (bornology α) :=
+{ le_Sup := λ S b hb,
+    by {rw [bornology.le_def], exact inf_le_of_right_le (Inf_le  $ mem_image_of_mem _ hb)},
+  Sup_le := λ S b hb,
+    by {simp_rw [bornology.le_def] at *,
+        exact le_inf b.le_cofinite (le_Inf $ λ f ⟨b', hb', hfb'⟩, hfb' ▸ hb b' hb')},
+  Inf_le := λ S b hb, by {rw [bornology.le_def], exact le_Sup (mem_image_of_mem _ hb)},
+  le_Inf := λ S b hb,
+    by {simp_rw [bornology.le_def] at *,
+        refine Sup_le (λ f ⟨b', hb', hfb'⟩, hfb' ▸ hb b' hb')},
+  le_top := λ b, by {rw [bornology.le_def], exact bot_le},
+  bot_le := λ b, by {rw [bornology.le_def], exact b.le_cofinite},
+  ..bornology.lattice,
+  ..bornology.has_Inf,
+  ..bornology.has_Sup,
+  ..bornology.has_bot,
+  ..bornology.has_top }
+
+lemma is_bounded_Inf_iff {B : set (bornology α)} :
+  @is_bounded _ (Inf B) s ↔ ∀ b ∈ B, @is_bounded _ b s :=
+begin
+  let b : bornology α := of_bounded' {t | ∀ b ∈ B, @is_bounded _ b t}
+    (λ b' hb', @is_bounded_empty _ b')
+    (λ t₁ h₁ t₂ h₁₂ b' hb', @is_bounded.subset _ b' _ _ (h₁ b' hb') h₁₂)
+    (λ t₁ h₁ t₂ h₂ b' hb', @is_bounded.union _ b' _ _ (h₁ b' hb') (h₂ b' hb'))
+    (λ x b' hb', @is_bounded_singleton _ b' x),
+  suffices : Inf B = b,
+  { rw [this, is_bounded_of_bounded_iff], refl },
+  refine le_antisymm _ (le_Inf $ λ b' hb', bornology.le_iff.mpr $
+    λ s hs, ((is_bounded_of_bounded_iff _).mp hs) b' hb'),
+  rw bornology.le_iff,
+  intros s hs,
+  rw [is_bounded_of_bounded_iff],
+  intros b' hb',
+  revert s hs,
+  rw ← bornology.le_iff,
+  exact Inf_le hb'
+end
+
+lemma is_bounded_infi_iff {ι : Type*} {B : ι → bornology α} :
+  @is_bounded _ (⨅ i, B i) s ↔ ∀ i, @is_bounded _ (B i) s :=
+by rw [infi, is_bounded_Inf_iff, forall_range_iff]
+
+lemma is_bounded_inf_iff :
+  @is_bounded _ (b₁ ⊓ b₂) s ↔ @is_bounded _ b₁ s ∧ @is_bounded _ b₂ s :=
+by rw [inf_eq_infi, is_bounded_infi_iff, bool.forall_bool, and_comm]
+
+lemma cobounded_antitone : antitone (@cobounded α) :=
+λ b₁ b₂, id
+
+lemma cobounded_strict_anti : strict_anti (@cobounded α) :=
+λ b₁ b₂, id
```

Comment:
Maybe this works?
```suggestion
lemma cobounded_antitone : antitone (@cobounded α) := antitone_id
lemma cobounded_strict_anti : strict_anti (@cobounded α) := strict_anti_id
```

Reply:
`antitone_id` doesn't exist, but `monotone_id` does work !

Reply:
Oh whoops, I will add it then.